Automattic/woocommerce-payments
# Do not allow guest users to buy subscriptions with WooPay Direct Checkout

Fixes #9791

#### Changes proposed in this Pull Request








This PR fixes guest users being able to buy subscriptions using WooPay via Direct Checkout.

#### Testing instructions





* Login on WooPay.
* As a guest user on merchant site, add a subscription to the cart.
* Access the cart page on blocks checkout.
* Click the `Proceed to Checkout` button.
* You should not be redirected to WooPay.
* Access the cart page on classic checkout.
* Click the `Proceed to checkout` button.
* You should not be redirected to WooPay.
* Remove the subscriptions from the cart and add other type of product.
* Access the cart page on blocks checkout.
* Click the `Proceed to Checkout` button.
* You should be redirected to WooPay.
* Return to merchant site.
* Access the cart page on classic checkout.
* Click the `Proceed to checkout` button.
* You should be redirected to WooPay.

-------------------


**Post merge**
